moshi/src/main/java/com/squareup/moshi/-MoshiKotlinExtensions.kt

```diff
@@ -32,11 +29,14 @@ import kotlin.reflect.typeOf
  * @return a [JsonAdapter] for [T], creating it if necessary. Note that while nullability of [T]
  *         itself is handled, nested types (such as in generics) are not resolved.
  */
-inline fun <reified T> Moshi.adapter(): JsonAdapter<T> {
-  return adapter(typeOf<T>())
-}
+@ExperimentalStdlibApi
+inline fun <reified T> Moshi.adapter(): JsonAdapter<T> = adapter(typeOf<T>())
+
+@ExperimentalStdlibApi
+inline fun <reified T> Moshi.Builder.addAdapter(adapter: JsonAdapter<T>): Moshi.Builder = add(typeOf<T>().toType(), adapter)
 
-inline fun <reified T> Moshi.Builder.addAdapter(adapter: JsonAdapter<T>) = add(typeOf<T>().toType(), adapter)
+@ExperimentalStdlibApi
+inline fun <reified T> Moshi.nextAdapter(skipPast: JsonAdapter.Factory, annotations: Set<Annotation>): JsonAdapter<T> = nextAdapter(skipPast, typeOf<T>().toType(), annotations)
```

Comment:
I actually wonder if this is possibly a footgun and should be left out. Often this is used in more abstract cases you don't actually know the type parameter and are just deferring to the `type` param. I spent a minute scratching my head at some test failures before realizing that it was because it was using a reified `Any` as the adapter to look up rather than the type.

Reply:
I vote to leave it out. We can always add, we can't remove.

Reply:
146780a